src/encoder.h

```diff
@@ -0,0 +1,38 @@
+#ifndef DD_OPENTRACING_ENCODER_H
+#define DD_OPENTRACING_ENCODER_H
+
+#include <datadog/opentracing.h>
+
+namespace datadog {
+namespace opentracing {
+
+// A function that encodes a collection of traces.
+typedef std::function<std::string(std::deque<Trace>)> TraceEncoder;
```

Comment:
Will there be other implementations? Why not just call msgpack::pack directly in HttpEncoder::payload?

Reply:
Then we could make HttpEncoder::payload static, and then we don't even need a class. What does envoy need, can we just give them a function or something?

Reply:
The things envoy needs are more-or-less the three functions from HttpEncoder.

(We want to still control the implementation details of those, it just needs the data)

I'll tidy this up, make it simpler / flatter.